### PR TITLE
added an exec query option for the sql-load to improve loading times

### PR DESCRIPTION
**Pull request by zJeremiah**

### CODE REVIEW

**Review by ryanpmortensen — APPROVED** [review]

File: apps/workers/sql-load/main.go [hunk 1]
```diff
@@ -16,7 +17,9 @@ type options struct {
 	Postgres bootstrap.DBOptions `toml:"postgres"`
 	MySQL    bootstrap.DBOptions `toml:"mysql"`
 
-	sqlDB    *sql.DB
+	sqlDB  *sql.DB
+	sqlxDB *sqlx.DB // used for running direct exec command
```
Comment (jbsmith7741) [inline apps/workers/sql-load/main.go:21]: Do we need to have two separate connections to the db?

File: apps/workers/sql-load/worker.go [hunk 2]
```diff
@@ -124,33 +125,69 @@ func (w *worker) DoTask(ctx context.Context) (task.Result, string) {
 
 	// read the files for loading, verify columns types
 	start := time.Now()
-	err = w.ReadFiles()
+	err = w.ds.ReadFiles(w.flist, w.fileOpts, w.Params.SkipErr)
 	if err != nil {
-		return task.Failed(errors.Wrap(err, "readfiles error"))
+		return task.Failed(fmt.Errorf("readfiles error %w", err))
```
Comment (jbsmith7741) [inline apps/workers/sql-load/worker.go:130]: You could use `task.Failf("readfiles error %w", err)`

File: apps/workers/sql-load/worker.go [hunk 3]
```diff
@@ -465,3 +501,94 @@ func (ds *DataSet) findInsertKey(name string) bool {
 	}
 	return false
 }
+
+func RandString(n int) string {
+	rand.Seed(time.Now().UTC().UnixNano())
```
Comment (jbsmith7741) [inline apps/workers/sql-load/worker.go:506]: you probably could move the seed as part of an init function 
``` go
func init() {
rand.Seed(time.Now().UTC().UnixNano())
}
```

File: apps/workers/sql-load/worker.go [hunk 4]
```diff
@@ -25,7 +27,7 @@ type InfoURI struct {
 	DeleteMap map[string]string `uri:"delete"`                // map used to build the delete query statement
 	FieldsMap map[string]string `uri:"fields"`                // map json key values to different db names
 	Truncate  bool              `uri:"truncate"`              // truncate the table rather than delete
-
+	ExecQuery bool              `uri:"exec_query"`            // this will attempt to load the query data though a temp table (postgres only)
```
Comment (jbsmith7741) [inline apps/workers/sql-load/worker.go:30]: is exec_query the right name for it? maybe exec_load or just exec

**Review by jbsmith7741 — APPROVED** [review]
should we rename the app to sql-loadx?

File: apps/workers/sql-load/main.go (hunk 1 above)
Comment (zJeremiah) [reply apps/workers/sql-load/main.go:21]: the sqlx just uses the existing sql connection. I'm not sure if that creates a separate connection pool, but I didn't want to break the batch loader.

File: apps/workers/sql-load/worker.go (hunk 2 above)
Comment (zJeremiah) [reply apps/workers/sql-load/worker.go:130]: The fmt.Errorf will wrap the given error if you use the %w

File: apps/workers/sql-load/worker.go (hunk 3 above)
Comment (zJeremiah) [reply apps/workers/sql-load/worker.go:506]: That would probably work as well. Is there any benefit to moving it?

**Pull request merged**